### PR TITLE
FEATURE: Webhook for user creation and approval

### PR DESCRIPTION
**Pull request by fantasticfears**

### CODE REVIEW

**Comment by coding-horror** [comment]
Excellent can you review @tgxworld?

File: app/models/web_hook.rb [hunk 1]
```diff
@@ -67,6 +67,15 @@ def self.enqueue_topic_hooks(topic, user)
       )
     end
   end
+
+  %i(user_created user_approved).each do |event|
+    DiscourseEvent.on(event) do |user|
+      WebHook.enqueue_hooks(:user_creation,
+                            user_id: user.id,
```
Comment (tgxworld) [inline app/models/web_hook.rb]: NITPICK: I usually prefer that we don't flush the parameters all the way even though we do it in our Ember code. The downside of flushing the parameters is that if you change a line that changes the number of spaces it is indented by, all the following lines have to be updated and that pollutes the commit history.

**Comment by tgxworld** [comment]
Merged with my changes in c463cf63d4a40375519af56dd01d22654c149063

**Comment by tgxworld** [comment]
Thanks @fantasticfears!

**Pull request closed without merging**